bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/api/BondHttpApi.java

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bondhome.internal.api;
+
+import static org.openhab.binding.bondhome.internal.BondHomeBindingConstants.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.InputStreamContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.bondhome.internal.BondException;
+import org.openhab.binding.bondhome.internal.handler.BondBridgeHandler;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+
+/**
+ * {@link BondHttpApi} wraps the Bond REST API and provides various low
+ * level function to access the device api (not cloud api).
+ *
+ * @author Sara Geleskie Damiano - Initial contribution
+ */
+@NonNullByDefault
+public class BondHttpApi {
+    private final Logger logger = LoggerFactory.getLogger(BondHttpApi.class);
+    private final BondBridgeHandler bridgeHandler;
+    private final HttpClientFactory httpClientFactory;
+    private Gson gson = new Gson();
+
+    public BondHttpApi(BondBridgeHandler bridgeHandler, final HttpClientFactory httpClientFactory) {
+        this.bridgeHandler = bridgeHandler;
+        this.httpClientFactory = httpClientFactory;
+    }
+
+    /**
+     * Gets version information about the Bond bridge
+     *
+     * @return the {@link BondSysVersion}
+     * @throws BondException
+     */
+    public BondSysVersion getBridgeVersion() throws BondException {
+        String json = request("/v2/sys/version");
+        logger.trace("BondHome device info : {}", json);
+        try {
+            return Objects.requireNonNull(gson.fromJson(json, BondSysVersion.class));
```

Comment:
`request()` can return empty string, in which case `fromJson()` will return null. This in turn will cause `NullPointerException` from `requireNonNull()`. I believe this is unhandled, so it will probably stop the binding from working. The same applies to the other methods changed in 53b977846a820ebe78af61ec2dd8ceedf15ed33b.

Reply:
ooh, actually we're already covered. in request(), it will throw a BondException if it doesn't look like JSON: https://github.com/openhab/openhab-addons/pull/13459/files#diff-d9fd10ab3e0d1d3d185e26f283666097886bdb069508f0440cc254ee3da6e0c0R241

this will cover an empty string.

Reply:
So last line in that method is unreachable? It explicitly returns an empty string.

Reply:
yes. I'll push another commit to make that more obvious